config/initializers/yaml_renderer.rb

```diff
@@ -1,4 +1,7 @@
 ActionController::Renderers.add :yaml do |obj, _|
-  data = JSON.load(obj.to_json).to_yaml
+  # Create a serializable resource instance
+  serializable_resource = ActiveModelSerializers::SerializableResource.new(obj)
+
+  data = JSON.parse(serializable_resource.to_json).to_yaml
```

Comment:
Can you use `data = serializable_resource.as_json.to_yaml` here? I don't like that this goes all the way into a string and then back and then into YAML.